utils/validate-quickstart-data-sources.ts

```diff
@@ -0,0 +1,80 @@
+import {
+  fetchPaginatedGHResults,
+  filterQuickstartConfigFiles,
+  filterOutTestFiles,
+  GithubAPIPullRequestFile,
+  isNotRemoved,
+} from './lib/github-api-helpers';
+
+import { getCoreDataSourceIds } from './lib/nr-graphql-helpers';
+
+import Quickstart from './lib/Quickstart';
+import DataSource from './lib/DataSource';
+import { passedProcessArguments } from './lib/helpers';
+
+export const validateDataSourceIds = async (
+  githubFiles: GithubAPIPullRequestFile[]
+) => {
+  const coreDataSourceIds = await getCoreDataSourceIds();
```

Comment:
We'll need to handle the error case for this function. Depending on what direction you choose, we could either add a try catch or just exit if no data source ids are returned.